File: library/src/main/java/androidz/util/ColorUtil.java
package androidz.util;

import static android.graphics.Color.TRANSPARENT;

import androidx.annotation.ColorInt;
import androidx.core.graphics.ColorUtils;

public class ColorUtil {

    public static boolean isDarkColor(@ColorInt int color) {
        return color != TRANSPARENT && ColorUtils.calculateLuminance(color) < 0.5;
    }

    public static boolean isLightColor(@ColorInt int color) {
        return color != TRANSPARENT && ColorUtils.calculateLuminance(color) > 0.5;
    }

}

File: library/src/main/java/androidz/util/ColorUtil.kt
package androidz.util

import android.graphics.Color
import androidx.annotation.ColorInt
import androidx.core.graphics.ColorUtils


object ColorUtil {

    @JvmStatic
    fun isDarkColor(@ColorInt color: Int): Boolean {
        return color != Color.TRANSPARENT && ColorUtils.calculateLuminance(color) < 0.5
    }

    @JvmStatic
    fun isLightColor(@ColorInt color: Int): Boolean {
        return color != Color.TRANSPARENT && ColorUtils.calculateLuminance(color) > 0.5
    }
}
